ambry-protocol/src/test/java/com/github/ambry/protocol/RequestResponseTest.java

```diff
@@ -672,6 +672,109 @@ public void replicaMetadataRequestTest() throws IOException {
     MessageInfoAndMetadataListSerde.AUTO_VERSION = oldMessageInfoVersion;
   }
```

Comment:
Testcase coverage seems minimal currently. Please add positive, negative, null check, exception, etc tests for all the classes and methods added.

Reply:
This is because we don't have multiple versions in above request objects.

Reply:
Added other checks for positive ,negative, null and exception.